freqtrade/freqai/data_kitchen.py

```diff
@@ -1188,7 +1287,91 @@ def use_strategy_to_populate_indicators(
         """
         Use the user defined strategy for populating indicators during retrain
         :param strategy: IStrategy = user defined strategy object
-        :param corr_dataframes: dict = dict containing the informative pair dataframes
+        :param corr_dataframes: dict = dict containing the df pair dataframes
+                                (for user defined timeframes)
+        :param base_dataframes: dict = dict containing the current pair dataframes
+                                (for user defined timeframes)
+        :param pair: str = pair to populate
+        :param prediction_dataframe: DataFrame = dataframe containing the pair data
+        used for prediction
+        :param do_corr_pairs: bool = whether to populate corr pairs or not
+        :return:
+        dataframe: DataFrame = dataframe containing populated indicators
+        """
+
+        # this is a hack to check if the user is using the populate_any_indicators function
+        new_version = inspect.getsource(strategy.populate_any_indicators) == (
+            inspect.getsource(IStrategy.populate_any_indicators))
+
+        if new_version:
+            tfs: List[str] = self.freqai_config["feature_parameters"].get("include_timeframes")
+            pairs: List[str] = self.freqai_config["feature_parameters"].get(
+                "include_corr_pairlist", [])
+
+            for tf in tfs:
+                if tf not in base_dataframes:
+                    base_dataframes[tf] = pd.DataFrame()
+                if not corr_dataframes.keys():
```

Comment:
Running live/dry with the "`new_version`" is throwing an `KeyError` for the `corr_dataframes` dict. I think it is due to this line. 

Here is the error:

```
  File "/home/rcaulk/projects/freqtrade/user_data/strategies/ReinforcementLearningExample5ac.py", line 134, in populate_indicators
    dataframe = self.freqai.start(dataframe, metadata, self)
  File "/home/rcaulk/projects/freqtrade/freqtrade/freqai/freqai_interface.py", line 142, in start
    dk = self.start_live(dataframe, metadata, strategy, self.dk)
  File "/home/rcaulk/projects/freqtrade/freqtrade/freqai/freqai_interface.py", line 421, in start_live
    dataframe = dk.use_strategy_to_populate_indicators(
  File "/home/rcaulk/projects/freqtrade/freqtrade/freqai/data_kitchen.py", line 1335, in use_strategy_to_populate_indicators
    dataframe = self.populate_features(dataframe.copy(), corr_pair, strategy,
  File "/home/rcaulk/projects/freqtrade/freqtrade/freqai/data_kitchen.py", line 1250, in populate_features
    informative_df = self.get_pair_data_for_features(
  File "/home/rcaulk/projects/freqtrade/freqtrade/freqai/data_kitchen.py", line 1201, in get_pair_data_for_features
    dataframe = corr_dataframes[pair][tf]
KeyError: '15m'
```

It is very odd, when I debug into this exact line from `freqai_interface.py`, I confirm that upon entry into this function, the `corr_dataframes` dict is not set to the default value (despite `freqai_interface.py line 421 not passing any key values for `corr_dataframes`). Instead, `corr_dataframes` does have the corr_pairlist keys, but only a single tf (`3m` in this case). However it is missing the higher `include_timeframes`. Since you have this line here, it is preventing the function from adding those keys. Which throws the error posted above. 

It seems you added this line for some fix in the bt tests? I guess I will let you decide how to ameliorate this since your head is deeper into the new backtesting updates.

Reply:
Fixed! Thanks